File: 2984486(small)/nicpe92/5634947029139456/0/extracted/Charge.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Charge {

	public static void main(String[] args) throws Exception{
		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
		int T =Integer.parseInt(in.readLine());
		for (int c = 0; c < T; c++) {
			String[] s = in.readLine().split(" " );
			int N = Integer.parseInt(s[0]);
			int L = Integer.parseInt(s[1]);
			int[] mat = new int[N];
			s =  in.readLine().split(" " );
			for (int i = 0; i < N; i++) {
				for (int j = 0; j < L; j++) {
					if(s[i].charAt(j)=='1')
					{
						mat[i] |= (1<<j);
					}
				}
			}
			int[] mat2 = new int[N];
			s =  in.readLine().split(" " );
			for (int i = 0; i < N; i++) {
				for (int j = 0; j < L; j++) {
					if(s[i].charAt(j)=='1')
						mat2[i] |= (1<<j);
				}
			}
			int min = 1000;
			a: for (int i = 0; i < (1<<L); i++) {
				int cnt = 0, j=i;
				while(j>0)
				{
					if(j%2==1)
						cnt++;
					j/=2;
				}
				boolean[] match = new boolean[N];
				int[] nuev = new int[N];
				for (int k = 0; k < N; k++) {
					for (int k2 = 0; k2 < L; k2++) {
						if((mat[k]&(1<<k2))!=0 && (i&(1<<k2))==0)
							nuev[k]|=(1<<k2);
						else if((mat[k]&(1<<k2))==0 && (i&(1<<k2))!=0)
							nuev[k]|=(1<<k2);
					}
					for (int k2 = 0; k2 < N; k2++) {
						if(nuev[k] == mat2[k2] && !match[k2])
						{
							match[k2]=true;
						}
					}
				}
				for (int k = 0; k < N; k++) {
					if(!match[k])
						continue a;
				}
				min = Math.min(min, cnt);
			}
			System.out.print("Case #"+(c+1)+": ");
			if(min < 1000)
				System.out.println(min);
			else System.out.println("NOT POSSIBLE");
		}
	}
}

File: 2984486(small)/MoOT.Cube/5634947029139456/0/extracted/Charge.scala
package charging

import collection.JavaConversions._

object Charge{
  def read(file: String) = {
    val s = io.Source.fromFile(file)
    val c = s.getLines.toList
    s.close
    c
  }

  def write(output: Seq[String], file: String){
    val buf = new java.io.BufferedWriter(new java.io.FileWriter(file))
    val res = output.zipWithIndex.map{case (v, i) => s"Case #${i+1}: $v"}.mkString("\n")
    println(res)
    buf.write(res)
    buf.close
  }

  def test(devices: Seq[(Int, Seq[Int])], toCharge: Seq[Int], i: Int, l: Int): Int = {
    val b = Integer.parseInt(("1" + "0"*i), 2)
    // println(b)
    if(i < l) test(
      devices ++ devices.map{case (_i, d) => (_i+1, d.map{_ ^ b}.sorted)}, toCharge, i+1, l)
    else {
      val m = devices.filter(_._2 == toCharge).map{_._1}
      if(m.size > 0) return m.min else -1
    }
  }

  def solve(input: Seq[String]): String = {
    val (n, l) = input.head.split(" ").toList match { case (n :: l :: _) => (n.toInt, l.toInt)}
    val devices = input.tail.head.split(" ").toSeq.map{Integer.parseInt(_, 2)}.sorted
    val toCharge = input.tail.tail.head.split(" ").toSeq.map{Integer.parseInt(_, 2)}.sorted
    // println(devices)
    // println(toCharge)
    val res = test(Seq((0, devices)), toCharge, 0, l)
    if(res == -1)"NOT POSSIBLE" else res.toString
  }

  def main(args: Array[String]){
    val input = read(args(0)).tail.grouped(3)
    write(input.map{solve}.toSeq, "outcharging")
  }
}
